stan/math/prim/functor/hcubature.hpp

```diff
@@ -0,0 +1,521 @@
+// For the code in hcubature.hpp, the package JuliaMath/HCubature.jl
+// written in Julia by Steven G. Johnson served as a template.
+// It comes with the following MIT "Expat" license:
+//
+// Copyright (c) 2017: Steven G. Johnson.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files
+// (the "Software"), to deal in the Software without restriction,
+// including without limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of the Software,
+// and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+
+#ifndef STAN_MATH_PRIM_FUNCTOR_HCUBATURE_HPP
+#define STAN_MATH_PRIM_FUNCTOR_HCUBATURE_HPP
+
+#include <set>
+#include <stan/math/prim/fun/choose.hpp>
+#include <stan/math/prim/fun/isfinite.hpp>
+#include <stan/math/prim/fun/fabs.hpp>
+#include <stan/math/prim/fun/fmax.hpp>
+#include <stan/math/prim/fun/max.hpp>
+
```

Comment:
We do includes from our files first, then outside libraries, then standard library

```suggestion
#include <stan/math/prim/fun/choose.hpp>
#include <stan/math/prim/fun/isfinite.hpp>
#include <stan/math/prim/fun/fabs.hpp>
#include <stan/math/prim/fun/fmax.hpp>
#include <stan/math/prim/fun/max.hpp>
#include <set>

```

Reply:
Comes in next commit.